ui/app/src/components/CRUDButton/CRUDButton.tsx

```diff
@@ -0,0 +1,65 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Button, Tooltip } from '@mui/material';
+import { OverridableStringUnion } from '@mui/types';
+import { ButtonPropsColorOverrides, ButtonPropsVariantOverrides } from '@mui/material/Button/Button';
+import { useIsReadonly } from '../../model/config-client';
+
+interface CRUDButtonProps {
```

Comment:
I'm wondering if this button should be moved to the package `component` instead. We probably want that also in the dashboard.
WDYT @eunicorn @sjcobb ?

Reply:
Of course if we want to move the button to the package component, we should add more parameter like `isReadonly` instead of using directly the function `useIsReadonly`

and yes you are right it achieve the same purpose. This wrapper of the MUI button just add on top of that the tooltip that advertise why the button is disable.